sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -560,44 +597,60 @@ object GpuCast extends Arm {
    * Asserts that all values in a column are within the specific range.
    *
    * @param values ColumnVector to be performed with range check
-   * @param minValue Named parameter for function to create Scalar representing range minimum value
-   * @param maxValue Named parameter for function to create Scalar representing range maximum value
+   * @param minValue Range minimum value of input type T
+   * @param maxValue Range maximum value of input type T
    * @param inclusiveMin Whether the min value is included in the valid range or not
    * @param inclusiveMax Whether the max value is included in the valid range or not
    * @param errorMsg Specify the message in the `IllegalStateException`
    * @throws IllegalStateException if any values in the column are not within the specified range
    */
-  private def assertValuesInRange(values: ColumnView,
-      minValue: => Scalar,
-      maxValue: => Scalar,
+  private def assertValuesInRange[T](values: ColumnView,
+      minValue: T,
+      maxValue: T,
       inclusiveMin: Boolean = true,
       inclusiveMax: Boolean = true,
-      errorMsg:String = GpuCast.OVERFLOW_MESSAGE): Unit = {
+      errorMsg:String = GpuCast.OVERFLOW_MESSAGE)
+      (implicit ord: Ordering[T]): Unit = {
 
-    def throwIfAny(cv: ColumnView): Unit = {
-      withResource(cv) { cv =>
-        withResource(cv.any()) { isAny =>
-          if (isAny.isValid && isAny.getBoolean) {
+    def throwIfAnyNan(): Unit = {
+      withResource(values.isNan()) { valuesIsNan =>
+        withResource(valuesIsNan.any()) { anyNan =>
+          if (anyNan.isValid && anyNan.getBoolean) {
             throw RapidsErrorUtils.arithmeticOverflowError(errorMsg)
           }
         }
       }
     }
 
-    withResource(minValue) { minValue =>
-      throwIfAny(if (inclusiveMin) {
-        values.lessThan(minValue)
-      } else {
-        values.lessOrEqualTo(minValue)
-      })
+    def throwIfOutOfRange(minInput: T, minValue: T,
+                        maxInput: T, maxValue: T): Unit = {
```

Comment:
I just realize that `minValue` and `maxValue` are given as the top function parameter thus we don't need them here again.
```suggestion
    def throwIfOutOfRange(minInput: T, maxInput: T): Unit = {
```